rules/TypeDeclaration/Rector/Class_/ReturnTypeFromStrictTernaryRector.php

```diff
@@ -68,119 +76,71 @@ public function getValue($number): int
     {
-        return [Class_::class];
+        return [ClassMethod::class, Function_::class, Closure::class];
     }
 
     /**
-     * @param Class_ $node
+     * @param ClassMethod|Function_|Closure $node
      */
     public function refactorWithScope(Node $node, Scope $scope): ?Node
     {
-        $hasChanged = false;
-
-        foreach ($node->getMethods() as $classMethod) {
-            if ($classMethod->returnType instanceof Node) {
-                continue;
-            }
-
-            $onlyStmt = $classMethod->stmts[0] ?? null;
-            if (! $onlyStmt instanceof Return_) {
-                continue;
-            }
-
-            if (! $onlyStmt->expr instanceof Ternary) {
-                continue;
-            }
-
-            $ternary = $onlyStmt->expr;
-
-            // has scalar in if/else of ternary
-            $ternaryIfElseTypes = $this->matchScalarTernaryIfElseTypes($ternary);
-            if (! $ternaryIfElseTypes instanceof TernaryIfElseTypes) {
-                continue;
-            }
-
-            $ifType = $ternaryIfElseTypes->getFirstType();
-            $elseType = $ternaryIfElseTypes->getSecondType();
+        if ($this->shouldSkip($node, $scope)) {
+            return null;
+        }
 
-            if (! $this->areTypesEqual($ifType, $elseType)) {
-                continue;
-            }
+        if ($node->stmts === null) {
+            return null;
+        }
 
-            if ($this->classMethodReturnTypeOverrideGuard->shouldSkipClassMethod($classMethod, $scope)) {
-                continue;
-            }
+        $returns = $this->betterNodeFinder->findInstanceOf($node->stmts, Return_::class);
+        if (count($returns) !== 1) {
+            return null;
+        }
 
-            $returnTypeNode = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode($ifType, TypeKind::RETURN);
-            if (! $returnTypeNode instanceof Node) {
-                continue;
-            }
+        $return = $returns[0];
+        if (! $return->expr instanceof Ternary) {
+            return null;
+        }
+        $ternary = $return->expr;
 
-            $classMethod->returnType = $returnTypeNode;
-            $hasChanged = true;
+        $nativeTernaryType = $scope->getNativeType($ternary);
+        if ($nativeTernaryType instanceof MixedType) {
+            return null;
         }
 
-        if ($hasChanged) {
-            return $node;
+        $ternaryType = $this->staticTypeMapper->mapPhpParserNodePHPStanType($ternary);
+        $returnTypeNode = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode($ternaryType, TypeKind::RETURN);
+        if (! $returnTypeNode instanceof Node) {
+            return null;
         }
 
-        return null;
+        $node->returnType = $returnTypeNode;
+        return $node;
     }
 
     public function provideMinPhpVersion(): int
     {
-        return PhpVersionFeature::SCALAR_TYPES;
+        return PhpVersionFeature::NULLABLE_TYPE;
```

Comment:
Php 7.0 for scalar feature should be ok, as far as return type validated via StaticTypeMapper which check feature type

Reply:
done :+1: